### PR TITLE
refactor: set removal version for `Deno.ListenTlsOptions.certFile`, `Deno.ListenTlsOptions.keyFile` and `Deno.ConnectTlsOptions.certFile`

### PR DESCRIPTION
**Pull request by iuioiua**
This change:
1. Sets the removal version for `Deno.ListenTlsOptions.certFile`, `Deno.ListenTlsOptions.keyFile` and `Deno.ConnectTlsOptions.certFile` for Deno v2, in favour of the `cert`, `key` and `caCerts` options, respectively.
2. Replaces use of the deprecated options with the new recommended options.

Towards #22021

### CODE REVIEW

File: ext/net/02_tls.js [hunk 1]
```diff
@@ -76,6 +76,20 @@ function listenTls({
   if (transport !== "tcp") {
     throw new TypeError(`Unsupported transport: '${transport}'`);
   }
+  if (keyFile !== undefined) {
+    internals.warnOnDeprecatedApi(
+      "Deno.ListenTlsOptions.keyFile",
+      new Error().stack,
+      "Pass the key to the `key` option instead.",
```
Comment (bartlomieju) [inline ext/net/02_tls.js:83]:
```suggestion
      "Pass the key file contents to the `Deno.ListenTlsOptions.key` option instead.",
```

File: ext/net/02_tls.js [hunk 2]
```diff
@@ -76,6 +76,20 @@ function listenTls({
...
+    );
+  }
+  if (certFile !== undefined) {
+    internals.warnOnDeprecatedApi(
+      "Deno.ListenTlsOptions.certFile",
+      new Error().stack,
+      "Pass the cert to the `cert` option instead.",
```
Comment (bartlomieju) [inline ext/net/02_tls.js:90]: Ditto

File: ext/net/02_tls.js [hunk 3]
```diff
@@ -92,6 +106,13 @@ async function startTls(
     alpnProtocols = undefined,
   } = {},
 ) {
+  if (certFile !== undefined) {
+    internals.warnOnDeprecatedApi(
+      "Deno.ListenTlsOptions.certFile",
```
Comment (bartlomieju) [inline ext/net/02_tls.js:111]: Ditto

**Review by bartlomieju — COMMENTED** [review]
Looks good with a small change suggested

File: ext/net/02_tls.js (hunk 3 above)
Comment (iuioiua) [reply ext/net/02_tls.js]: Actually, I removed this change. See #22032.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**